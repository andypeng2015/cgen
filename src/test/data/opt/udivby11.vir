module foo

export function w $foo(w %x) {
@start:
  %x = udiv.w %x, 11_w
  ret %x
}
