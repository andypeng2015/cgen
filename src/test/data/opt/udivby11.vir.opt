module foo

export function w $foo(w %x) {
@2:
  %1 = umulh.w %x, 0x1745d175_w ; @5
  ret %1
}
