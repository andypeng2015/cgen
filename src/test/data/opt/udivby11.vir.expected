module foo

export function w $foo(w %x) {
@0:
  %0 = umulh.w %x, 0x1745d175_w ; @2
  ret %0
}
